include/dgl/base_heterograph.h

```diff
@@ -547,6 +550,25 @@ DGL_DEFINE_OBJECT_REF(FlattenedHeteroGraphRef, FlattenedHeteroGraph);
 HeteroGraphPtr CreateHeteroGraph(
     GraphPtr meta_graph, const std::vector<HeteroGraphPtr>& rel_graphs);
 
+/*!
+ * \brief Create a heterograph from meta graph and a list of bipartite graph,
+ * additionally specifying number of nodes per type.
+ */
+HeteroGraphPtr CreateHeteroGraph(
+    GraphPtr meta_graph,
+    const std::vector<HeteroGraphPtr>& rel_graphs,
+    const std::vector<int64_t>& num_nodes_per_type);
+
+HeteroGraphPtr CreateHeteroGraph(
+    GraphPtr meta_graph,
+    const std::vector<HeteroGraphPtr>& rel_graphs,
+    std::vector<int64_t>&& num_nodes_per_type);
+
+HeteroGraphPtr CreateHeteroGraph(
+    GraphPtr meta_graph,
+    const std::vector<HeteroGraphPtr>& rel_graphs,
+    IdArray num_nodes_per_type);
```

Comment:
The second one is weird to me. Right value reference is for mutating the argument w/ no copy which quite contradicts with the one with `const vector<>&`. In terms of using vector or IdArray, I think picking either one is fine. My practice is using IdArray if the value will participate in FFI. So my suggestion is using IdArray.

Reply:
We can also extend the original interface as this right?
```c++
HeteroGraphPtr CreateHeteroGraph(
    GraphPtr meta_graph,
    const std::vector<HeteroGraphPtr>& rel_graphs,
    const IdArray num_nodes_per_type = aten::NullArray());
```

Reply:
I don't think r-value reference is for mutating the argument (for which we should always use pointers).  The r-value reference here is for implementing moves, i.e. replacing `num_verts_per_type_` member with the given one without any copying.
Also, IdArray would be converted to `std::vector` anyway because the member `num_verts_per_type_` is a `std::vector`.